pylammpsmpi/wrapper/extended.py

```diff
@@ -228,3 +233,97 @@
     "cellbeta",
     "cellgamma",
 ]
+
+
+class LammpsLibrary:
+    """
+    Top level class which manages the lammps library provided by LammpsBase
+    """
+
+    def __init__(
+        self,
+        cores=1,
+        oversubscribe=False,
+        enable_flux_backend=False,
+        working_directory=".",
+        client=None,
+        mode="local",
+        cmdargs=None,
+    ):
+        self.cores = cores
+        self.working_directory = working_directory
+        self.oversubscribe = oversubscribe
+        self.enable_flux_backend = enable_flux_backend
+        self.client = client
+        self.mode = mode
+
+        if self.mode == "dask":
+            fut = self.client.submit(
+                LammpsBase,
+                cores=self.cores,
+                oversubscribe=self.oversubscribe,
+                working_directory=self.working_directory,
+                cmdargs=cmdargs,
+                actor=True,
+            )
+            self.lmp = fut.result()
+
+            fut = self.lmp.start_process()
+            _ = fut.result()
```

Comment:
@srmnitc This part is still confusing to me, basically you send the `LammpsBase` class to `dask` initialize it on a remote worker and then send the initialized `LammpsBase` instance back to the main process.

Reply:
This is correct, this is the [Dask Actors](https://distributed.dask.org/en/latest/actors.html#:~:text=Actors%20enable%20stateful%20computations%20within,methods%20on%20that%20remote%20object.) feature. The returned object is an Actor, which can be used to execute methods on the version of the class which is submitted to the remote worker.

Reply:
> This is correct, this is the [Dask Actors](https://distributed.dask.org/en/latest/actors.html#:~:text=Actors%20enable%20stateful%20computations%20within,methods%20on%20that%20remote%20object.) feature. The returned object is an Actor, which can be used to execute methods on the version of the class which is submitted to the remote worker.

That's an interesting concept, I guess we could use this for the workflows as well, @liamhuber what do you think? Currently this is not yet supported by the standard `concurrent.futures.Executor`, but it is definitely something we should keep in mind.

Reply:
Our current "best practices" is for the node behaviour to be fully functional, and it's this functional method of the node object that we ship off to the executor, along with arguments. In the case, I suppose it's possible that such an `Actor` paradigm allows some optimization in what's getting serialized and shipped around? I'm not yet familiar enough with parallelization in python to have a good gut sense.

The other issue is that while "functional" nodes are best practice, we do _allow_ passing `self` as part of the node function. I haven't done extensive tests for asynchronous operation yet, but I guess it's possible that we _need_ either and `Actor` paradigm, or to return and parse `self` in the case that `self` is needed for the task. Again, I'm still quite inexperience on parallel execution, but since it's an edge case (non-functional) on top of something fancy (parallel execution), I probably won't dig deeper into it for a couple weeks yet...